### DISCUSSION TITLE
Generation of Optionset values limited to 1033 in  TS-Typing generator

### DISCUSSION THREAD

**Discussion by cyco77** [Power-Maverick/DataverseDevTools-VSCode#112] [General]
Is there a reason why the generation is limited to language 1033?
![image](https://github.com/Power-Maverick/DataverseDevTools-VSCode/assets/1198698/9d98b345-916e-4221-8a56-7620ff5e43c7)

May be someone want's to create his files in another language.

**Comment by Power-Maverick** [comment]
Many languages can be installed on a single environment. So when you are generating you get array of UserLocalized labels. I won't know which to generate hence I defaulted to 1033. I need to check if multiple languages exists what is the 0th index.

**Comment by cyco77** [comment]
right now, you take the userlocalized language. That means when I change my settings from english to german, I get my german translations. I tried it after removing the language check and the generator works like a charme :) in german, but it works...

**Comment by Power-Maverick** [comment]
I get your point. But what happens when you have multiple languages installed?

**Comment by cyco77** [comment]
with my current pr the generator uses the selected user language. same functionality as in 

English
![image](https://github.com/Power-Maverick/DataverseDevTools-VSCode/assets/1198698/21bfac03-5446-4f1e-a6e7-2bc75549df43)
German:
![image](https://github.com/Power-Maverick/DataverseDevTools-VSCode/assets/1198698/6343902b-5a14-4aee-9371-acee8b815c2b)
